### PR TITLE
Recorded Future for Splunk SOAR 4.3.3 - Bug fixes - Alert Polling

### PR DESCRIPTION
**Pull request by JQvickWester**
Please ensure your pull request (PR) adheres to the following guidelines:

- Please refer to our contributing documentation for any questions on submitting a pull request, link: [Contribution Guide](https://github.com/splunk-soar-connectors/.github/blob/main/.github/CONTRIBUTING.md)

#### Please check if your PR fulfills the following requirements:
- [X] Testing of all the changes has been performed (for bug fixes / features)
- [X] The manual_readme_content.md has been reviewed and added / updated if needed (for bug fixes / features)
- [X] Use the following format for the PR description: `<App Name>: <PR Type> - <PR Description>`
- [X] Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.
- [ ] Verify all checks are passing.
- [X] Do NOT use the `next` branch of the forked repo. Create separate feature branch for raising the PR.
- [X] Do NOT submit updates to dependencies unless it fixes an issue.
#### Please check the type of change your PR introduces:
- [ ] New App
- [X] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation
- [ ] Other (please describe):

## Security Considerations (REQUIRED)
- [ ] If you are exposing any endpoints using a [REST handler](https://docs.splunk.com/Documentation/SOAR/current/DevelopApps/RESTHandlers),
  please document them in the `manual_readme_content.md`.
- [ ] If this is a new connector or you are adding new actions
    - [ ] Please document in the `manual_readme_content.md` all methods (eg, OAuth) used to authenticate
      with the service that the connector is integrating with.
    - [ ] If any actions are unable to run on SOAR Cloud, please document this in the `manual_readme_content.md`.
- [ ] Are you introducing any new cryptography modules? If yes, please elaborate their purpose:
- [ ] Are you are accessing the file system? If yes, please verify that you are only accessing paths returned through
the [Vault](https://docs.splunk.com/Documentation/SOARonprem/5.2.1/DevelopApps/AppDevAPIRef#Vault) API.
- [ X ] Are you are marking code to be ignored by Semgrep with [`nosemgrep`](https://semgrep.dev/docs/ignoring-files-folders-code/#ignoring-code-through-nosemgrep)?
If yes, please provide justification in an additional comment next to the ignored code.

## Release Notes (REQUIRED)
**Unreleased**

- Changes to polling of alerts; now requires a comma separated list to pull in alerts.
- Fixing logic issue blocking the polling of alerts.
- Fixing issues with hardcoded path for Cloud, instead utilizing Vault class to access path. 

## What is the current behavior? (OPTIONAL)
- Describe the current behavior that you are modifying.

## What is the new behavior? (OPTIONAL)
- Describe the behavior or changes that are being added by this PR.

## Other information (OPTIONAL)
- Any other information that is important to this PR such as screenshots of how the component looks before and after the change.

## Pay close attention to (OPTIONAL)
- Any specific code change or test case points which must be addressed/reviewed at the time of GA release.

## Screenshots (if relevant)

---
Thanks for contributing!

### CODE REVIEW

**Comment by mishalp-crest** [comment]
@JQvickWester  
We usually update the app's minor version when there is a bug fix. Let us know if you want to update it.

**Comment by mishalp-crest** [comment]
@JQvickWester Can you please confirm?

**Comment by nplebanskyi** [comment]
Hello @mishalp-crest, I'm colleague of @JQvickWester, he's currently on long vacation. Why did you change the publisher from `Recorded Future, Inc` to `Splunk` [in this commit](https://github.com/splunk-soar-connectors/recordedfuture/pull/22/commits/c79e4feee6b81c4765505afc48066ab60b024393)? And also why did you decrease the `min_phantom_version` in the same commit?

File: recordedfuture.json [hunk 1]
```diff
@@ -1,17 +1,17 @@
 {
     "appid": "6efe0e1b-76ac-4ffd-8fa0-ac58fd6efd77",
     "name": "Recorded Future For Splunk SOAR",
-    "description": "This app implements investigative actions to perform lookups for quick reputation information, contextual threat intelligence and external threat alerts",
-    "publisher": "Recorded Future, Inc",
+    "description": "Enhance your security posture with Recorded Future for Splunk SOAR.\nKey Capabilities:\n\u2022Swift Threat Assessments: Access Recorded Future's extensive IOC data for swift and accurate assessments",
+    "publisher": "Splunk",
     "type": "reputation",
     "main_module": "recordedfuture_connector.py",
-    "app_version": "4.3.2",
```
Comment (nplebanskyi) [inline recordedfuture.json:8]: @mishalp-crest, the patch version was update by @JQvickWester here. Is there some other file where the version needs to be updated? Thanks.

Comment (mishalp-crest) [reply recordedfuture.json:8]: @nplebanskyi Whenever a new action is added to the app, the major version of the app should be updated as per standard practice 
That is the concern

Comment (nplebanskyi) [reply recordedfuture.json:8]: I see your point, yes, then it makes sense to update the minor version, thank you.

File: README.md [hunk 2]
```diff
@@ -2,11 +2,11 @@
 # Recorded Future For Splunk SOAR
 
 Publisher: Recorded Future, Inc  
-Connector Version: 4.3.2  
+Connector Version: 4.3.3
```
Comment (nplebanskyi) [inline README.md:5]:
```suggestion
Connector Version: 4.4.0
```

File: recordedfuture.json [hunk 3]
```diff
@@ -1,11 +1,11 @@
 {
     "appid": "6efe0e1b-76ac-4ffd-8fa0-ac58fd6efd77",
     "name": "Recorded Future For Splunk SOAR",
-    "description": "This app implements investigative actions to perform lookups for quick reputation information, contextual threat intelligence and external threat alerts",
+    "description": "Enhance your security posture with Recorded Future for Splunk SOAR.\nKey Capabilities:\n\u2022Swift Threat Assessments: Access Recorded Future's extensive IOC data for swift and accurate assessments",
     "publisher": "Recorded Future, Inc",
     "type": "reputation",
     "main_module": "recordedfuture_connector.py",
-    "app_version": "4.3.2",
+    "app_version": "4.3.3",
```
Comment (nplebanskyi) [inline recordedfuture.json:8]:
```suggestion
    "app_version": "4.4.0",
```

File: recordedfuture_consts.py [hunk 4]
```diff
@@ -20,11 +20,11 @@
 # and limitations under the License.
 
 # Define your constants here
-version = "4.3.2"
-buildid = "331"
+version = "4.3.3"
```
Comment (nplebanskyi) [inline recordedfuture_consts.py:23]:
```suggestion
version = "4.4.0"
```

**Comment by nplebanskyi** [comment]
@mishalp-crest could you please apply my suggestions for version and also revert the changes to `min_phantom_version` again? It should be `6.2.1` and not `6.1.1`. Thank you!

**Comment by mishalp-crest** [comment]
@nplebanskyi Done with the changes
let us know if you have any concern / doubt

**Comment by nplebanskyi** [comment]
@mishalp-crest all looks good, thank you.

**Review by nplebanskyi — APPROVED** [review]

**Pull request merged**